akka-docs/src/main/paradox/typed/persistence-durable-state.md

```diff
@@ -0,0 +1,391 @@
+
+More effects are explained in @ref:[Effects and Side Effects](#effects-and-side-effects).
+
+In addition to returning the primary `Effect` for the command `DurableStateBehavior`s can also 
+chain side effects that are to be performed after successful persist which is achieved with the `thenRun`
+function e.g @scala[`Effect.persist(..).thenRun`]@java[`Effect().persist(..).thenRun`].
+
+### Completing the example
+
+Let's fill in the details of the example.
+
+Commands:
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #command }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #command }
+
+State is a storage for the latest value of the counter.
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #state }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #state }
+
+The command handler handles the commands `Increment`, `IncrementBy` and `GetValue`. 
+
+* `Increment` increments the counter by `1` and persists the updated value as an effect in the State
+* `IncrementBy` increments the counter by the value passed to it and persists the updated value as an effect in the State
+* `GetValue` retrieves the value of the counter from the State and replies with it to the actor passed in
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #command-handler }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #command-handler }
+
+@scala[These are used to create a `DurableStateBehavior`:]
+@java[These are defined in an `DurableStateBehavior`:]
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #behavior }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #behavior }
+
+## Effects and Side Effects
+
+A command handler returns an `Effect` directive that defines what state, if any, to persist. 
+Effects are created using @java[a factory that is returned via the `Effect()` method] @scala[the `Effect` factory]
+and can be one of: 
+
+* `persist` will persist state atomically, i.e. all states are stored or none of them are stored if there is an error
```

Comment:
Not sure about all states, is it not the case that you always persist one state (with durable state)?

Reply:
corrected ..